pkg/vault/contracts/Vault.sol

```diff
@@ -1522,6 +1550,14 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
         return amount + _MAX_CONVERT_ERROR;
     }
 
+    // Minimum swap amount (applied to scaled18 values), enforced as a security measure to block potential
+    // exploitation of rounding errors
+    function _ensureValidTradeAmount(uint256 tradeAmount) private pure {
+        if (tradeAmount < _MINIMUM_TRADE_AMOUNT && tradeAmount != 0) {
```

Comment:
How about checking `!= 0` first? super minor, but that's going to be the case pretty often in unbalanced swaps.

Reply:
I had it like that and switched it to this, thinking most trades would be swaps (and thus non-zero). How many straight swaps do we have vs. unbalanced add/remove liquidity operations?

Reply:
The difference is that in a successful swap you will still need to check both conditions, whereas in a successful `add` / `remove` you won't.

We don't care optimizing gas for cases that will revert anyways.